src/components/VirtualizedTreeSelect.js

```diff
@@ -103,15 +109,36 @@ class VirtualizedTreeSelect extends Component {
     if (inputValue.length === 0) {
       return !option.parent || this.data[option.parent].expanded;
     } else {
-      if (candidate.label.toLowerCase().indexOf(inputValue)) {
-        return true;
+      return option.visible;
+    }
+
+  }
+
+  _filterValues(searchInput) {
+    const matches = []
+    for (let option of this.state.options) {
+      if (option[this.props.labelKey].toLowerCase().indexOf(searchInput) !== -1) {
```

Comment:
Should lowercase `searchInput` as well, otherwise this does not work (tested with demo). Also, I suggest extracting the match checking into a separate function, as we might want to extend the component API with support for custom matching functions (providing e.g. fuzzy string matching).

Reply:
I completely missed that, Thank you! The match checking in a separate function is a great idea, gonna implement that for sure 👨‍💻